### PR TITLE
Added feature to remove duplicated images and update database

### PR DESCRIPTION
**Pull request by peterjaap**

### CODE REVIEW

**Comment by peterjaap** [comment]
This PR also includes the changes in #7 and #8

**Comment by peterjaap** [comment]
My local test;

```
8228 duplicated files have been deleted
0 rows have been updated in the catalog_product_entity_varchar table
26333 rows have been updated in the catalog_product_entity_media_gallery table
764 Mb has been freed.
```

File: Service/DuplicateFileFinder.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * Adapted from https://gist.github.com/vchenin/35e71c4e65cc8d9da64b37853811e9c7
+ */
+namespace Sivaschenko\CleanMedia\Service;
+
+class DuplicateFileFinder {
```
Comment (sivaschenko) [inline Service/DuplicateFileFinder.php]: Can you please format the code o match Magento standards and be consistent for better readability

(probably I have to add some kind of static tests to the repo for objectivity))

File: Command/CatalogMedia.php [hunk 2]
```diff
@@ -143,14 +186,61 @@ protected function execute(InputInterface $input, OutputInterface $output)
         }
 
         if ($input->getOption(self::INPUT_KEY_REMOVE_ORPHANED_ROWS)) {
-            $this->resource->getConnection()->delete($this->resource->getTableName(Gallery::GALLERY_TABLE), ['value IN (?)' => $missingFiles]);
+            $db->delete($this->resource->getTableName(Gallery::GALLERY_TABLE), ['value IN (?)' => $missingFiles]);
+        }
+
+        $duplicatedFiles = [];
+        if ($input->getOption(self::INPUT_KEY_LIST_DUPES)) {
```
Comment (sivaschenko) [inline Command/CatalogMedia.php]: The method becomes too long. I'd extract this complex part of logic to a separate class

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for all the contributions @peterjaap ! Please see my improvement suggestions.

**Comment by peterjaap** [comment]
@sivaschenko could you take a look at this PR again please?

**Comment by sivaschenko** [comment]
Refactored the implementation and introduced a separate command for duplicates.
Need to test the PR and update README before merging

**Comment by peterjaap** [comment]
@sivaschenko nice, LGTM! :+1:

**Pull request merged**